src/inc/engine/engine.h

```diff
@@ -13,22 +15,28 @@ class Engine
 {
 public:
 
+  Engine( std::vector<std::unique_ptr<GameComponent>>& components );
+
   void initialize( int height, int width );
 
-  void advance( std::vector<std::unique_ptr<GameComponent>>& components );
+  void advance();
 
   bool peek_has_updated();
 
   GameRenderer& get_renderer();
 
 private:
 
-  void render( std::vector<std::unique_ptr<GameComponent>>& components );
+  void render_components();
+
+  void update_components( InputEvent& input_event );
 
-  void update( std::vector<std::unique_ptr<GameComponent>>& components );
+  InputEvent& process_input();
```

Comment:
Method to handle all miscellaneous engine stuff before handing event off to components